### PR TITLE
V4-Java-SDK-PPPLMER-94838-update query parameters-PART A

### PR DESCRIPTION
**Pull request by saikgupta**
This is for ticket PPPLMER-94838[https://engineering.paypalcorp.com/jira/browse/PPPLMER-94838](url). Please review the changes done for missing query parameters for below resources:
1. List Bank Account.
2. List Bank Card.
3. List Paypal Account.
4. List Venom Account.

### CODE REVIEW

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccountTest.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package com.hyperwallet.clientsdk.model;
 
+import com.hyperwallet.clientsdk.model.HyperwalletVenmoAccount.Type;
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccountTest.java:3]: Unused import. Please remove all the unused imports in all the files.

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java [hunk 2]
```diff
@@ -774,7 +775,7 @@ public HyperwalletBankCard createBankCard(HyperwalletBankCard bankCard) {
             throw new HyperwalletException("Bank Card token may not be present");
         }
         if (bankCard.getType() == null) {
-            bankCard.setType(HyperwalletTransferMethod.Type.BANK_CARD);
+            bankCard.setType(Type.BANK_CARD);
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java:778]: Change it to HyperwalletBankCard.Type.VENMO_ACCOUNT for better clarity.

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 3]
```diff
@@ -3777,6 +3777,30 @@ public void testListTransfer_withListOptions() throws Exception {
             Mockito.any(TypeReference.class));
     }
 
+
+    @Test
+    public void testListPayPalAccount_withSomeParameters() throws Exception {
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:3782]: Change testName with actual parameter name

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPayPalAccountTest.java [hunk 4]
```diff
@@ -13,7 +13,7 @@ protected HyperwalletPayPalAccount createBaseModel() {
         payPalAccount
                 .token("test-token")
                 .type(HyperwalletTransferMethod.Type.PAYPAL_ACCOUNT)
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPayPalAccountTest.java:15]: Do we have any separate ticket to change HyperwalletTransferMethod.Typeto HyperwalletPayPalAccount.Type?

**Review by ramahalingam — CHANGES_REQUESTED** [review]
Pls tag the JIRA ticket id in the PR comments.

**Comment by saikgupta** [comment]
@ramahalingam Changes done as per your review comments.

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletBankCardTest.java [hunk 5]
```diff
@@ -12,9 +15,9 @@ protected HyperwalletBankCard createBaseModel() {
         HyperwalletBankCard bankCard = new HyperwalletBankCard();
         bankCard
                 .token("test-token")
-                .type(HyperwalletTransferMethod.Type.PREPAID_CARD)
+                .type(Type.BANK_CARD)
```
Comment (Rrathinasabapath) [inline src/test/java/com/hyperwallet/clientsdk/model/HyperwalletBankCardTest.java:18]: Please remove the following blank line

**Review by Rrathinasabapath — CHANGES_REQUESTED** [review]
Please remove the blank line in HyperwalletBankCardTest.java

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java (hunk 3 above)
Comment (saikgupta) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:3782]: done

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java (hunk 2 above)
Comment (saikgupta) [reply src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java]: done

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccountTest.java (hunk 1 above)
Comment (saikgupta) [reply src/test/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccountTest.java]: done

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPayPalAccountTest.java (hunk 4 above)
Comment (saikgupta) [reply src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPayPalAccountTest.java]: done

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletBankCardTest.java (hunk 5 above)
Comment (saikgupta) [reply src/test/java/com/hyperwallet/clientsdk/model/HyperwalletBankCardTest.java:18]: done

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java [hunk 6]
```diff
@@ -846,6 +847,7 @@ public HyperwalletBankCard updateBankCard(HyperwalletBankCard bankCard) {
         String url = paginate(this.url + "/users/" + userToken + "/bank-cards", options);
         if (options != null) {
             url = addParameter(url, "status", options.getStatus());
+
```
Comment (Rrathinasabapath) [inline src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java:850]: Pls remove the blank line

**Review by Rrathinasabapath — APPROVED** [review]
Approving the changes

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPayPalAccountTest.java (hunk 4 above)
Comment (ramahalingam) [reply src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPayPalAccountTest.java]: Why this line is removed completely?

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccount.java [hunk 7]
```diff
@@ -15,9 +15,11 @@
 @XmlAccessorType(XmlAccessType.FIELD)
 public class HyperwalletVenmoAccount extends HyperwalletBaseMonitor {
 
+    public enum Status {ACTIVATED, INVALID, VERIFIED,DE_ACTIVATED}
+    public enum Type {VENMO_ACCOUNT}
```
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccount.java:19]: why do we need this to be public, I mean does this class represents another type in the future? `HyperwalletVenmoAccount` is always `VENMO_ACCOUNT` account right? please double check other FI objects as well

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccount.java [hunk 8]
```diff
@@ -48,16 +50,16 @@ public HyperwalletVenmoAccount clearToken() {
         return this;
     }
 
-    public HyperwalletTransferMethod.Type getType() {
+    public Type getType() {
         return type;
     }
 
-    public void setType(HyperwalletTransferMethod.Type type) {
+    public void setType(Type type) {
```
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccount.java:57]: if the type is for this class `HyperwalletVenmoAccount` is always `VENMO_ACCOUNT` why do we allow users of this model to modify it?

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccount.java (hunk 7 above)
Comment (peter-joseph) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccount.java:19]: moreover `HyperwalletVenmoAccount` type is always `VENMO_ACCOUNT` would it be safe and sound right that everytime we instantiate this construct the instance will always be of type `VENMO_ACCOUNT` ?

**Review by peter-joseph — COMMENTED** [review]
I have made an example review on FI type Venmo `HyperwalletVenmoAccount` please consider also updating other FI that share same pattern.. we don't want the users of hyperwallet model to be confused of the types that each instance represent, if we needed the type then we have to make sure that each instance for example `HyperwalletVenmoAccount` will only represent 1 type.

Cheers,

Peter

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java [hunk 9]
```diff
@@ -1604,7 +1606,7 @@ public HyperwalletVenmoAccount createVenmoAccount(HyperwalletVenmoAccount venmoA
             throw new HyperwalletException("Venmo Account token may not be present");
         }
         if (venmoAccount.getType() == null) {
```
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java:1608]: if we fix our models, we won't be doing any of this extra work

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccount.java (hunk 7 above)
Comment (saikgupta) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletVenmoAccount.java:19]: Hi Peter, There are other resources like Bank Account which has multiple type like :
public static enum Type {BANK_ACCOUNT, WIRE_ACCOUNT}, and also previously these resources were taking the type from Transfer method which had all the Type clubbed together, so we created separate type in each resources and Setter method is required for the create resource API call...
public class HyperwalletTransferMethod extends HyperwalletBaseMonitor {

    public enum Type {BANK_ACCOUNT, WIRE_ACCOUNT, PREPAID_CARD, BANK_CARD, PAPER_CHECK, PAYPAL_ACCOUNT, VENMO_ACCOUNT}

**Review by vkorotchenko — APPROVED** [review]

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java (hunk 2 above)
Comment (ramahalingam) [reply src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java:790]: Delete this line.

**Review by ramahalingam — COMMENTED** [review]
Pls fix the comment.

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java (hunk 2 above)
Comment (ramahalingam) [reply src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java:790]: Delete this line

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java [hunk 10]
```diff
@@ -774,7 +786,9 @@ public HyperwalletBankCard createBankCard(HyperwalletBankCard bankCard) {
             throw new HyperwalletException("Bank Card token may not be present");
         }
         if (bankCard.getType() == null) {
-            bankCard.setType(HyperwalletTransferMethod.Type.BANK_CARD);
+
+            bankCard.setType(Type.BANK_CARD);
+            bankCard.setType(HyperwalletBankCard.Type.BANK_CARD);
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java:791]: Delete the line : bankCard.setType(Type.BANK_CARD);

**Review by ramahalingam — APPROVED** [review]

**Comment by Rrathinasabapath** [comment]
Merging the changes

**Pull request merged**